torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -1111,34 +1105,16 @@ void CudaCodeGen::call(const std::vector<CallArg>& args) {
     }
   }
 
-  // Bind the buffer addresses into arguments
-  auto const& buffer_args = this->buffer_args();
   int ptr_count = buffer_args.size();
-  if (has_random_) {
-    ptr_count += 2;
-  }
-  std::vector<void*> args_data(buffer_args.size());
   std::vector<void*> ptr_to_args(ptr_count);
+  for (size_t i = 0; i < buffer_args.size(); i++) {
+    ptr_to_args[i] =
+        buffer_args[i].isVar() ? raw_args[i] : const_cast<void**>(&raw_args[i]);
+  }
   uint64_t rand_seed = uint64_t(-1);
   uint64_t rand_offset = uint64_t(-1);
-  for (size_t i = 0; i < buffer_args.size(); i++) {
-    auto const& bufferArg = buffer_args[i];
-    if (bufferArg.isVar()) {
-      auto stype = bufferArg.dtype().scalar_type();
-      switch (stype) {
-#define TYPE_CASE(Type, Name)             \
-  case ScalarType::Name:                  \
-    ptr_to_args[i] = args[i].Name##Ptr(); \
-    break;
-        AT_FORALL_SCALAR_TYPES_AND2(Bool, Half, TYPE_CASE);
-#undef TYPE_CASE
-        default:
-          throw unsupported_dtype();
-      }
-    } else {
-      args_data[i] = args[i].data();
-      ptr_to_args[i] = &args_data[i];
-    }
+  if (has_random_) {
+    ptr_count += 2;
```

Comment:
Oh, I think I forgot to move it with the other chunk of code, it should be before the line 1109. Basically, if we have a rand, we need to add two extra arguments.

Thanks for catching!

Reply:
Add a test for this case, if possible?

Reply:
I'm pretty sure we already have tests for this, since this PR just moves this code around. I'm not sure why it passed, maybe because of undefined behavior (we overflowed the buffer, but it still worked)

Reply:
Yeah, I was wondering the same as well.